compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java

```diff
@@ -504,16 +503,17 @@ private static void populateImmutableStructureFields(Types types, SymbolTable sy
                                                         unresolvedTypes);
 
             Name origFieldName = origField.name;
-            BVarSymbol immutableFieldSymbol = new BVarSymbol(origField.symbol.flags | flag,
-                                                             origFieldName, pkgID, immutableFieldType,
-                                                             immutableStructureSymbol, origField.pos, SOURCE);
+            BInvokableSymbol immutableFieldSymbol = new BInvokableSymbol(origField.symbol.tag,
```

Comment:
Is this the correct fix? I don't think using BInvokableSymbol is correct when the type of the field is not a function type.

You can check with something like
```ballerina
type Foo record {|
    int a;
    function () b;
    int|function () c;
|};
```